src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6073,4 +6073,25 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="IDS_FeatureModuleInitializers" xml:space="preserve">
     <value>module initializers</value>
   </data>
+  <data name="ERR_ModuleInitializerMethodMustBeAccessibleOutsideTopLevelType" xml:space="preserve">
+    <value>Module initializer method '{0}' must be accessible outside top-level type '{1}'</value>
```

Comment:
>must be accessible outside top-level type '{1} [](start = 43, length = 46)

Why not say "on the module level", or "from any type in this program"?

Reply:
"On the module level" sounds less clear to me, and "from any type in this program" sounds less direct. ("Why is just any type in this program involved?") That's my reasoning, but I don't feel strongly about anything.

Reply:
If I was the one running afoul of this check, I would appreciate the attention to detail that gets me thinking in the most immediately helpful terms. I really do like putting an instantly recognized name in the message rather than only providing jargon to the developer. The runtime cost during compilation is negligible and the extra code needed probably already lives somewhere else. Is there some way we could address your concern while still achieving some of the same goals?

Reply:
> If I was the one running afoul of this check, I would appreciate the attention to detail that gets me thinking in the most immediately helpful terms. I really do like putting an instantly recognized name in the message rather than only providing jargon to the developer. The runtime cost during compilation is negligible and the extra loop needed probably already lives somewhere else. Is there a way we could address your concern while still achieving some of the same goals?

I do not believe we are going to get enough value back by spending implementation/testing/review efforts on this aspect. No code, no cost.

---
In reply to: [418363820](https://github.com/dotnet/roslyn/pull/43301#discussion_r418363820) [](ancestors = 418363820)

Reply:
My disappointment is a bit keen, perhaps because of how nebulous my understanding is of how much effort that could be, but I'll accept your judgment.

Reply:
@AlekseyTs In honesty, it bothers me that the improved message is being dismissed as too much effort when it is already written and constituted of a single loop that there is probably already a helper for. We're talking about this code which is about to get even simpler now that it only has this one caller:

```cs
var topLevelType = containingTypes(this).Last();

static IEnumerable<NamedTypeSymbol> containingTypes(Symbol symbol)
{
    for (var type = symbol.ContainingType; type is { }; type = type.ContainingType)
    {
        yield return type;
    }
}
```

Yet in the previous PR, you asked why we didn't just go ahead and implement VB because "it is not that hard to do and no good reason not to." The same applies to the work I've already done which comprises a single loop, and I've provided good reasons to do it.

Reply:
@jnm2 

> "On the module level" sounds less clear to me,

It may be less clear but it's also more correct. This feature is a module level feature, not acknowledging that in the error is likely to create as much confusion in some cases as hiding it.

Reply:
Good, thanks! What do you think of 'at' instead of 'on'?

> Module initializer method '{0}' must be accessible at the module level

Reply:
> Good, thanks! What do you think of 'at' instead of 'on'?

Sounds fine to me